### DISCUSSION TITLE
Fetch via URL blocked on Vercel --> NextJS (API)

### DISCUSSION THREAD

**Discussion by teyo5** [vercel/community#492] [Help]
I am trying to fetch a csv file via URL (nextjs api route) and while it works locally, the deployed version on vercel seem to block it. To be precise I am using d3 ( `d3.csv()` ) in the api route and it fails with the following error:

error:
`Access to fetch at 'https://xx.vercel.app/api/xx?id=1Or7N2' from origin 'https://xx-xxxgh7-txxo5.vercel.app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

api/xx (using [d3](https://github.com/d3/d3):)
```
const key = push(setRef).key;
    if (key) {
      const storageURL: string =
        'https://firebasestorage.googleapis...';

      await csv(storageURL)  // --> this fails
        .then(async data => {
          ...
```

called like this:
```
 const refreshData = () => {
    replace(asPath);
  };

  const addGuess = async () => {
    await fetch(`${server}/api/xx?id=${currentUser.uid}`, {
      method: 'POST',
    });
    refreshData(); // --> probably not relevant
  };
```

I really hope someone can help. It's for college and I have to hand it in on thursday 🥲
Thanks in advance!

**Comment by samcx** [comment]
Hi there,

We apologize we couldn't help sooner! Were you able to resolve your issue?
